docs/requirements.txt

```diff
@@ -14,21 +15,18 @@ joblib
 lxml
 matplotlib
 numpy
-pip
 pykml>=0.2
 pyproj
 scikit-image
 scipy
-# pyaps dependencies
-cdsapi
-ecCodes
-netcdf4
-pygrib
-# pyhdf  # required by MERRA, which is currently not supported in pyaps3
-# pyresample dependencies
+pysolid
+pyaps3
+# for ARIA, FRInGE, HyP3, GMTSAR
+gdal>=3
```

Comment:
`gdal` is still not so easy to install and is not required if one is not using the above InSAR products. Since `requirements.txt` is meant for installing mintpy to an existing env, I would prefer keeping gdal out of the `requirements.txt` file, for more flexibility.

It's easy to add gdal in the command line while running conda install if one wants to.

Reply:
:+1: makes sense to me -- I'll leave it here, but commented out.

Reply:
Well, I commented it out, but because CircleCI uses `requirements.txt` and the `SanFranSenDT42` test needs GDAL, I had to add it to the CicleCI config. 

While `gdal` can be a rather large pain, in general installing it via conda is pretty painless at this point, so I don't see a big harm including it. 

I'm not terribly invested either way though as I will almost always use the `environment.yml` -- whichever way you want to go is fine by me.